### DISCUSSION TITLE
Salvar página em PDF e enviar por e-mail via Javascript

### DISCUSSION THREAD

**Discussion by carlosmassam** [frontendbr/forum#1738] [Perguntas e Respostas]
Na empresa que eu trabalho tem um portal onde as pessoas podem ver seu espelho ponto.
A situação é o seguinte: Os colaboradores acessam o portal, selecionam os dias que precisam que seja corrigido o ponto e clicam num botão onde é gerado uma nova página, com uns input onde então é preenchido os horários corretos e o motivo da correção.
Acontece que essa segunda página que gera é somente visual, as pessoas preenchem os campos só que os valores não são armazenados em lugar nenhum,  então depois imprimem e passam a impressão para seus superiores.
A minha dúvida é: Existe a possibilidade de gerar um PDF desta pagina exatamente como ela está (Como se fosse imprimir em PDF) e depois enviar para um email via javascript?

estou colocando uns prints pra ficar mais facil de entender.

1) Parte onde seleciona os dias que precisam justificar os pontos
[https://uploaddeimagens.com.br/imagens/zpkeQM0](https://uploaddeimagens.com.br/imagens/zpkeQM0)

2) Página que gera com os dados selecionados. Então é preenchidos os campos e impresso. É desta página que eu preciso que seja gerado um PDF e seja enviado por e-mail.
[https://uploaddeimagens.com.br/imagens/nRWFBng](https://uploaddeimagens.com.br/imagens/nRWFBng)

Desde já, eu agradeço. Obrigado.

**Comment by tresvezesheber** [comment]
Saudações, @carlosmassam! Existe uma lib, em javascript, chamada jsPDF que lhe permite gerar documentos em PDF via client-side. Creio que consigas gerar os PDF com ela... A parte do envio por e-mail eu já não tenho conhecimento suficiente para lhe ajudar.

**jsPDF**:
Página no [GitHub](https://github.com/MrRio/jsPDF) e [documentação](https://rawgit.com/MrRio/jsPDF/master/docs/index.html).

**Comment by matheus-vieira** [comment]
@carlosmassam posso tentar contribuir com a parte de envio do pdf após gerado.

Existem serviços que disponibilizam o envio de email.

[Form Spree](https://formspree.io/) e [Form Submit](https://formsubmit.co/) são exemplos desses serviços..

No meu [site pessoal](https://matheus-vieira.github.io/contact/) eu utilizava o Form Spree, porém ele alterou algumas coisa como o redirecionamento e me obrigou a buscar um concorrente, o Form Submit resolveu esse redirecionamento.

Outro exemplo de utilização é nesse meu projeto de [Solicitação de compra](https://matheus-vieira.github.io/solicitacao-compra/) que está disponível no github

**Comment by PaulinoBrunoMar** [comment]
@carlosmassam conseguiu realizar essa feature, estou buscando a mesma solução?
